Source/Base/Base/Math/Interpolation.h

```diff
@@ -0,0 +1,403 @@
+                {
+                    ClampAnim(t);
+                    if (t == 0.0f || t == 1.0f)
+                        return t;
+
+                    f32 c5 = (2.0f * Math::PI) / 4.5f;
+                    if (t < 0.5f)
+                        return -(pow(2.0f, 20.0f * t - 10.0f) * Math::Sin((20.0f * t - 11.125f) * c5)) / 2.0f;
+
+                    return (pow(2.0f, -20.0f * t + 10.0f) * Math::Sin((20.0f * t - 11.125f) * c5)) / 2.0f + 1.0f;
+                }
+
+                static f32 Expo(f32 t)
+                {
+                    ClampAnim(t);
+                    if (t == 0.0f || t == 1.0f)
+                        return t;
+
+                    if (t < 0.5f)
+                        return pow(2.0f, 20.0f * t - 10.0f) / 2.0f;
+
+                    return (2 - pow(2.0f, -20.0f * t + 10.0f)) / 2.0f;
+                }
+
+                static f32 Back(f32 t)
+                {
+                    f32 c1 = 1.70158f;
+                    f32 c2 = c1 * 1.525f;
+
+                    if (t < 0.5f)
+                        return (pow(2.0f * t, 2.0f) * ((c2 + 1.0f) * 2.0f * t - c2)) / 2.0f;
+
+                    return (pow(2.0f * t - 2.0f, 2.0f) * ((c2 + 1.0f) * (t * 2.0f - 2.0f) + c2) + 2.0f) / 2.0f;
+                }
+
+                static f32 Bounce(f32 t)
+                {
+                    ClampAnim(t);
+                    if (t < 0.5f)
+                        return (1.0f - Ease::Out::Bounce(1.0f - 2.0f * t)) / 2.0f;
+
+                    return (1.0f + Ease::Out::Bounce(2.0f * t - 1.0f)) / 2.0f;
+                }
+
+                static f32 Polynomial(f32 t, i32 degree)
+                {
+                    ClampAnim(t);
+                    ClampPolynomialDegree(degree);
+
+                    if (t < 0.5f)
+                        return pow(2.0f, static_cast<f32>(degree - 1)) * Ease::In::Polynomial(t, degree);
+
+                    return 1.0f - pow(-2.0f * t + 2.0f, static_cast<f32>(degree)) / 2.0f;
+                }
+
+                static f32 Quadratic(f32 t) { return Polynomial(t, 2); }
+                static f32 Cubic(f32 t)     { return Polynomial(t, 3); }
+                static f32 Quartic(f32 t)   { return Polynomial(t, 4); }
+                static f32 Quintic(f32 t)   { return Polynomial(t, 5); }
+                static f32 Sextic(f32 t)    { return Polynomial(t, 6); }
+                static f32 Septic(f32 t)    { return Polynomial(t, 7); }
+                static f32 Octic(f32 t)     { return Polynomial(t, 8); }
+            };
+        };
+
+        struct Bezier
+        {
+            static vec3 Cubic(f32 t, const vec3& p0, const vec3& p1, const vec3& p2, const vec3& p3)
+            {
+                f32 t2 = t * t;
+                f32 t3 = t2 * t;
+
+                f32 h0 = -1.0f * t3 + 3.0f * t2 - 3.0f * t + 1.0f;
+                f32 h1 = 3.0f * t3 - 6.0f * t2 + 3.0f * t;
+                f32 h2 = -3.0f * t3 + 3.0f * t2;
+                //f32 h3 = t3;
```

Comment:
Left over code? If it needs to be there and also be a comment, please describe why it is there.

Reply:
This is just implicit, I put it in a comment because i want to keep it to remember it, but doesn't want to allocate a new variable, can I just add `//f32 h3 = t3; This is implicit` or you want I remove the comment entirely ?